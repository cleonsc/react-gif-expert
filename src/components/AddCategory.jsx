import React, { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState(''); //Esto no puede ser undefined

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  }

  const onFormSubmit = (event) => {
    event.preventDefault();
    const newCategory = inputValue.trim();
    
    if (newCategory.length <= 1) return;

    // De esta forma obtengo las categorias anteriores y hago el spread en un nuevo array
    // setCategories(categories => [...categories, inputValue])
    onNewCategory(newCategory);

    //Limpio el input
    setInputValue('');
  }

  return (
    <form onSubmit={onFormSubmit}>
      <input
        type="text"
        placeholder=" Buscar gif"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  )
}
