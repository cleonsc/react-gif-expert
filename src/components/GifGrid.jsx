import React from 'react'
import { getGifs } from '../helpers/getGifts';

export const GifGrid = ({ category }) => {
  
  getGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <p>Hola Mundo</p>
    </>
  )
}
