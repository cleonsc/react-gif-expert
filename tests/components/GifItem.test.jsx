import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas <GifItem/>', () => {
  const title = 'GetItem title';
  const url = 'http://www.test-git-item.com/';

  test('it should match with the snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  })

  test('it should show the image with the passed URL', () => {
    render(<GifItem title={title} url={url} />);

    // screen.debug(); //para evaluar lo que podemos evaluar
    // expect(screen.getByRole('img').src).toBe(url); // Esta forma es muy larga cuando se quiera evaluar muchas propiedades

    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('it should show the title in the components', () => {
    render(<GifItem title={title} url={url} />);
    // console.log(screen.getByText('GetItem title')); //devuelve true si encuenta la frase
    expect(screen.getByText(title)).toBeTruthy();
  });
});