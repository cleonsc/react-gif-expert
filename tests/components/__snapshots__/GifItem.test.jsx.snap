// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifItem/> it should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="GetItem title"
      src="http://www.test-git-item.com/"
    />
    <p>
      GetItem title
    </p>
  </div>
</div>
`;
